### PR TITLE
Remove the ability to import executors from plugins

### PR DESCRIPTION
**Pull request by kaxil**
Executors should no longer be registered or imported via Airflow's plugin mechanism -- these types of classes are just treated as plain python classes by Airflow, so there is no need to register them with Airflow.

We had already removed it from our docs: https://airflow.apache.org/docs/apache-airflow/stable/authoring-and-scheduling/plugins.html#plugins

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by jedcunningham — APPROVED** [review]

**Review by jscheffl — COMMENTED** [review]
I think we have a still ongoing discussion, not voted towards future scope of provders package. I think this discussion should settle first.

It would also mean that if this PR is merged, that the executor descriptions in provider packages need to be removed - or at least be deprecated until Airflow 2.x support is generated.

File: newsfragments/43289.significant.rst [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Support for adding executors via Airflow Plugins is removed
+
+Executors should no longer be registered or imported via Airflow's plugin mechanism -- these types of classes
+are just treated as plain python classes by Airflow, so there is no need to register them with Airflow.
```
Comment (uranusjr) [inline newsfragments/43289.significant.rst]:
```suggestion
are just treated as plain Python classes by Airflow, so there is no need to register them with Airflow.
```

**Comment by kaxil** [comment]
> I think we have a still ongoing discussion, not voted towards future scope of provders package. I think this discussion should settle first.
> 
> It would also mean that if this PR is merged, that the executor descriptions in provider packages need to be removed - or at least be deprecated until Airflow 2.x support is generated.

@jscheffl The change in this PR just impact importing executors "via plugins": https://airflow.apache.org/docs/apache-airflow/stable/authoring-and-scheduling/plugins.html#plugins

It is not removing executors from Providers

File: newsfragments/43289.significant.rst (hunk 1 above)
Comment (kaxil) [reply newsfragments/43289.significant.rst]: Fixed

**Pull request merged**